tutorials/nlp/Dialogue.ipynb

```diff
@@ -0,0 +1,5269 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jaosjY4rGRNH"
+      },
+      "source": [
+        "# Installing NeMo from source"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "goQzOSflEq27",
+        "outputId": "25afeb5c-38c6-47aa-88c8-29918f0696be"
+      },
+      "outputs": [],
+      "source": [
+        "\"\"\"\n",
+        "You can run either this notebook locally (if you have all the dependencies and a GPU) or on Google Colab.\n",
+        "\n",
+        "Instructions for setting up Colab are as follows:\n",
+        "1. Open a new Python 3 notebook.\n",
+        "2. Import this notebook from GitHub (File -> Upload Notebook -> \"GITHUB\" tab -> copy/paste GitHub URL)\n",
+        "3. Connect to an instance with a GPU (Runtime -> Change runtime type -> select \"GPU\" for hardware accelerator)\n",
+        "4. Run this cell to set up dependencies.\n",
+        "\"\"\"\n",
+        "\n",
+        "import os \n",
+        "# install NeMo\n",
+        "BRANCH = 'dialogue_state_tracking_refactor' #'main'\n",
+        "\n",
+        "!apt-get update && apt-get install -y libsndfile1 ffmpeg\n",
+        "!git clone https://github.com/NVIDIA/NeMo --branch dialogue_state_tracking_refactor\n",
+        "os.chdir('NeMo')\n",
+        "!./reinstall.sh\n",
+        "os.chdir('..')\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "GjQ_z_xQMDIb"
+      },
+      "source": [
+        "# Overview\n",
+        "\n",
+        "There are three tasks as part of this tutorial\n",
+        "\n",
+        "1. Intent and Slot Classification using Assistant Dataset and a BERT model\n",
+        "2. Intent Classification using Schema Guided Dialogue Dataset and a GPT2 model\n",
+        "3. Answer Extender using MS Marco NLGen Dataset and a BART model\n",
+        "\n",
+        "Feel free to skip to the task that interests you most after installing NeMo from source."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "AS-zwy8tEq2_"
+      },
+      "source": [
+        "# Task Description: Intent and Slot Classification using Assistant Dataset\n",
```

Comment:
Since this is extremely similar to the standalone intent/slot tutorial, I think it's worth explaining what the difference is between your implementation of it and the existing one in the "original" NeMo. In other words why would someone want to use your part of the library to train this model instead of the existing implementation.

Reply:
Done!